### PR TITLE
Add RandomApply and RandomChoice preprocessing layers

### LINKED ISSUE #20727: add RandomApply and RandomChoice
This was included in keras-cv. 

- https://github.com/keras-team/keras-cv/pull/434
- https://github.com/keras-team/keras-cv/pull/1979 [random-apply](https://github.com/keras-team/keras-cv/blob/94b0a551d03ee9de3d81663e2c1a680cb113f7f1/keras_cv/src/layers/preprocessing/random_apply.py#L22)

### PR DESCRIPTION
**Pull request by timon0305**
## Description

Closes #20727

Adds two new composition preprocessing layers ported from keras-cv:

### `keras.layers.RandomApply`
Wraps a single layer and applies it with a given probability during training. At inference, input is passed through unchanged.

```python
layer = keras.layers.RandomApply(
    layer=keras.layers.RandomFlip("horizontal"),
    rate=0.5,
)
```

### `keras.layers.RandomChoice`
Takes a list of layers and randomly selects one to apply during training. At inference, input is passed through unchanged.

```python
layer = keras.layers.RandomChoice(
    layers=[
        keras.layers.RandomFlip("horizontal"),
        keras.layers.RandomRotation(0.1),
        keras.layers.RandomZoom(0.1),
    ],
)
```

Both layers:
- Support serialization (`get_config` / `from_config`)
- Work with dict inputs (images, labels, bounding_boxes, etc.)
- Are safe to use in `tf.data` / `grain` pipelines
- Validate constructor arguments
- Support the `seed` parameter for reproducibility

## Files Changed
- `keras/src/layers/preprocessing/random_apply.py` (new)
- `keras/src/layers/preprocessing/random_choice.py` (new)
- `keras/src/layers/__init__.py` (import)
- `keras/api/layers/__init__.py` (API export)
- `keras/api/_tf_keras/keras/layers/__init__.py` (tf_keras API export)

### CODE REVIEW

File: keras/src/layers/preprocessing/random_apply.py [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+        return cls(**config)
+
+    def get_config(self):
+        config = {
+            "layer": serialization_lib.serialize_keras_object(self._layer),
+            "rate": self.rate,
+            "seed": self.seed,
+            "name": self.name,
+        }
+        return config
```
[the remark this answers is not in the document]
Comment (jeffcarp) [reply keras/src/layers/preprocessing/random_apply.py:95]: Please fix

File: keras/src/layers/preprocessing/random_choice.py [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+
+        seed_generator = self.generator
+        idx = backend.random.randint(
+            shape=(),
+            minval=0,
+            maxval=len(self._layers_list),
+            seed=seed_generator,
+        )
+        idx = int(idx)
+        layer = self._layers_list[idx]
```
[the remark this answers is not in the document]
Comment (jeffcarp) [reply keras/src/layers/preprocessing/random_choice.py:90]: Please fix

**Comment by hertschuh** [comment]
This PR is in Draft, let me know when it's ready for review.
